tests/utils.py

```diff
@@ -16,14 +28,58 @@ class Server:
     Represents the server we're testing against.
     """
 
-    def __init__(self, host: str, port: int) -> None:
+    def __init__(
+        self,
+        app: Callable,
+        host: str,
+        port: int,
+        bind: str = None,
+        certfile: str = None,
+        keyfile: str = None,
+    ) -> None:
+        self._app = app
         self._host = host
         self._port = port
+        self._config = hypercorn_config.Config()
+        self._config.bind = [bind or f"{host}:{port}"]
+        self._config.certfile = certfile
+        self._config.keyfile = keyfile
+        self._config.worker_class = "asyncio"
+        self._started = False
+        self._should_exit = False
 
     @property
     def netloc(self) -> Tuple[bytes, int]:
         return (self._host.encode("ascii"), self._port)
 
     @property
     def host_header(self) -> Tuple[bytes, bytes]:
-        return (b"host", self._host.encode("utf-8"))
+        return (b"host", self._host.encode("ascii"))
```

Comment:
Small unrelated change as a cleanup for #204 - for correctness we ought to decode this as `ascii`, I think?

Reply:
@florimondmanca 

Once I've got a strange error with the `Server` fixture:
https://gist.github.com/cdeler/d4b32031687c3a81f5b904c580cf56f0

_Just for your information_

Reply:
@cdeler Yup, I can see where that's coming from: we are replacing the `headers` request argument (that contains the host header) with the response `headers` (same variable name).

Is the code for the stack trace in master already, or is it in this PR? If it's in master, we can go for a fix. :)

Reply:
The code is from `master`

Reply:
I can create an issue